src/repo/repo.service.ts

```diff
@@ -85,7 +85,7 @@ export class RepoService {
     return item;
   }
 
-  async findOneByOwnerAndRepo(owner: string, repo: string): Promise<DbRepoWithStats> {
+  async findOneByOwnerAndRepo(owner: string, repo: string, range?: number): Promise<DbRepoWithStats> {
```

Comment:
Could just set a default here instead of doing several `||` checks on the optional (defaults should also be optional)

```suggestion
  async findOneByOwnerAndRepo(owner: string, repo: string, range: number = 30): Promise<DbRepoWithStats> {
```